tools/generators/epic-generator/index.ts

```diff
@@ -0,0 +1,186 @@
+  }
+
+  return {
+    repository: schema.repository,
+    title: schema.title,
+    message: schema.message || '*Description to be added*',
+  };
+}
+
+const checkAuthentication = async () => {
+  const authStatus = execSync(`gh auth status`).toString();
+
+  if (authStatus.search('You are not logged into any GitHub hosts') > 0) {
+    throw new Error('You are not logged into GitHub CLI (gh).');
+  }
+};
+
+interface Package {
+  name: string;
+  folder: string;
+  owners?: string[];
+}
+
+interface Ownership {
+  path: string;
+  owners: string[];
+}
+
+interface MigrationIssue {
+  issueUrl?: string;
+  assignee?: string;
+  packages: Package[];
+}
+
+interface MigrationIssueMap {
+  [key: string]: MigrationIssue;
+}
+
+const getConvergencePackages = (tree: Tree) => {
+  const projects = getProjects(tree);
+
+  let convergencePackages: Package[] = [];
+  projects.forEach((project, key) => {
+    if (project.projectType === 'library' && project.tags?.some(tag => tag === 'vNext')) {
+      convergencePackages.push({
+        name: key,
+        folder: project.root,
+      });
+    }
+  });
+
+  return convergencePackages;
+};
+
+const getCodeowners = (): Ownership[] => {
+  const codeownersContent = readFileSync('.github/CODEOWNERS', 'utf8');
+
+  return codeownersContent.split('\n').map(line => {
+    const [path, ...owners] = line.split(' ');
+
+    return {
+      path: path.replace(/\/+$/, ''),
+      owners,
+    };
+  });
+};
+
+const mapOwnerships = (packages: Package[], ownerships: Ownership[]): Package[] => {
+  return packages.map(pckg => {
+    const owners = ownerships.find(ownership => ownership.path === pckg.folder)?.owners || [];
+
+    return {
+      ...pckg,
+      owners,
+    };
+  });
+};
+
+const getPackages = (tree: Tree) => {
+  const packages = getConvergencePackages(tree);
+
+  const ownerships = getCodeowners();
+
+  return mapOwnerships(packages, ownerships);
+};
+
+const createEpic = (repo: string, title: string, message: string) => {
+  const issueUrl = execSync(`gh issue create --repo "${repo}" --title "${title}" --body "${message}"`)
+    .toString()
+    .trim();
+
+  return issueUrl;
+};
+
+const createIssue = (repo: string, issue: MigrationIssue, templateTitle: string) => {
+  const title = `${templateTitle} - ${issue.assignee || 'unknown'}`;
+  const message = stripIndents`
+    🚧 This is an auto-generated issue to individually track migration progress.
+
+    ### Packages to migrate:
+    ${issue.packages.map(pckg => `- ${pckg.name}`).join('\n')}
+  `;
+
+  let command = `gh issue create --repo "${repo}" --title "${title}" --body "${message}"`;
+
+  const issueUrl = execSync(command).toString().trim();
+
+  return issueUrl;
+};
+
+const generateIssues = (repo: string, templateTitle: string, packages: Package[]) => {
+  const migrationIssues = packages.reduce<MigrationIssueMap>((acc, pckg) => {
+    const teamOwner = pckg.owners!.find(owner => owner.startsWith('@microsoft/'));
```

Comment:
instead of using the non-null assertion operator maybe create a new type for package with owners ?

Reply:
This was an oversight from when I had `owners` as `owner?: string`, as I made the assumption that there would only be one owner. I've made a refactor to correct this.